src/modules/decklink/producer_decklink.cpp

```diff
@@ -853,7 +853,15 @@ mlt_producer producer_decklink_init( mlt_profile profile, mlt_service_type type,
 	// If allocated and initializes
 	if ( decklink && !mlt_producer_init( producer, decklink ) )
 	{
-		if ( decklink->open( arg? atoi( arg ) : 0 ) )
+		// Extract resource (card) from arg, removing path prefix, if any.
+		// (modules such as melted may pass arg with root_dir prefix)
+		char *arg_dup = strdup( arg ? arg : "" );
```

Comment:
I do not think a strdup() is necessary, and you can default (when arg == NULL) to "0" thereby removing the strlen() and free() lines below.